### DISCUSSION TITLE
Edge attribute dimension with GINEConv

### DISCUSSION THREAD

**Discussion by dpinski1** [pyg-team/pytorch_geometric#2599] [Q&A] [answered]
Hi, I'm trying to train a model using `GINEConv` that incorporates edge attributes. I'm confused by the second assert statement in the models `forward` method:

```
if isinstance(edge_index, Tensor):
            assert edge_attr is not None
            assert x[0].size(-1) == edge_attr.size(-1)
```

Let's say my node feature matrix X is of dimensions (10, 100) - 10 nodes with 100 features each. If there are 50 edges and I am only using one edge attribute, then my edge attribute matrix will be size 50. I don't understand why the size of my first row of the node feature matrix has to be equal to the size of my edge attribute matrix - is this claiming there should be a relationship between the number of node features and the number of edge attributes? Hoping someone can point out where I'm getting confused - thanks!

**Comment by rusty1s** [comment, marked as answer]
Yes, exactly. Node feature dimensions and edge feature dimensions have to match, as `GINEConv` utilizes summation to integrate edge features into source node features. A good way to allow that is to transform your edge features first before passing it into `GINEConv`, e.g.:
```python
x = conv(x, edge_index, lin(edge_attr))
```
where `lin` denotes a `Linear` layer mapping to the number of node features.

**Reply by dpinski1** [reply]
Thank you!! A follow up - in this case, should the parameters for the `Linear` layer be learnable? I'm assuming its fine if they are fixed.

**Reply by rusty1s** [reply]
I think it's reasonable to let `Linear` be trainable, as the model can learn how to integrate edge features to the message passing phase this way.
